domains.js

```diff
@@ -0,0 +1,72 @@
+const google = require('googleapis');
+const sheets = google.sheets('v4');
+
+function getDomains(username) {
+    const credentials = credentialConfiguration();
+    const request = {
+        spreadsheetId: credentials.TASK_SHEET_ID,
+        range: 'A:ZZZ', // extreme range to get whole spreadsheet
+        auth: credentials.API_KEY
+    };
+
+    return new Promise((resolve, reject) => {
+        sheets.spreadsheets.values.get(request, function(err, response) {
+            if (err) {
+                reject(err);
+            } else {
+                const domains = findMatch(username, response.values);
+                if (domains) {
+                    resolve({ 'domains': domains })
+                } else {
+                    reject({ 'error': `${username} not found.`})
+                }
+            };
+        });
+    });
+}
+
+function findMatch(username, records) {
+    let domains = records.filter(record => {
+        return username.toLowerCase() === record[0].toLowerCase()
+    })[0];
```

Comment:
> `array.prototype.find` is not supported by IE and a couple mobile browsers, notably Android.

Yeah, but this is Node.js code. You don’t have to worry about browsers at all here :)

(As a side note, `find()` is pretty easy to polyfill and I would generally recommend that for performance reasons in browser code, too.)